pyfar/classes/transmission_matrix.py

```diff
@@ -0,0 +1,670 @@
+
+        This should not be used directly. Instead use :py:func:`~from_tmatrix`.
+
+        """
+        shape = np.shape(data)
+        n_dim = len(shape)
+        if n_dim < 3 or shape[-3] != 2 or shape[-2] != 2:
+            raise ValueError("'data' must have a shape like "
+                             "(..., 2, 2, n_bins), e.g. (2, 2, 100).")
+
+        super().__init__(data, frequencies, comment)
+
+    @classmethod
+    def from_tmatrix(cls, data, frequencies, comment = ""):
+        """Create TransmissionMatrix using data in T-matrix shape
+        and frequencies.
+
+        For using individual objects for A, B, C, D matrix entries,
+        see :py:func:`~from_abcd` method.
+
+        Parameters
+        ----------
+        data : array_like, double
+            Raw data in the frequency domain.
+            In contrast to the :py:class:`~pyfar.classes.audio.FrequencyData`
+            class, data must have a shape of the form (..., 2, 2, N), e.g.
+            (2, 2, 1024) or (3, 2, 2, 1024). In those examples 1024 refers to
+            the number of frequency bins and the two dimensions before that to
+            the (2x2) ABCD matrices. Supported data types are ``int``,
+            ``float`` or ``complex``, where ``int`` is converted to ``float``.
+        frequencies : array_like, double
+            Frequencies of the data in Hz. The number of frequencies must match
+            the size of the last dimension of data.
+        comment : str, optional
+            A comment related to the data. The default is ``""``, which
+            initializes an empty string.
+
+        Examples
+        --------
+        >>> import pyfar as pf
+        >>> frequencies = (100,200,300)
+        >>> data = np.ones( (2, 2, len(frequencies)) )
+        >>> tmat = pf.TransmissionMatrix.from_tmatrix(data, frequencies)
+
+        >>> data = np.ones( (3, 2, 2, len(frequencies)) )
+        >>> tmat = pf.TransmissionMatrix.from_tmatrix(data, frequencies)
+
+        """
+        return cls(data, frequencies, comment)
+
+    @classmethod
+    def from_abcd(cls, A, B, C, D, frequencies = None):
+        """Create a TransmissionMatrix object from A-, B-, C-, D-data, and
+        frequencies.
+
+        Parameters
+        ----------
+        A, B, C, D : FrequencyData, array_like, double
+            Raw data for the matrix entries A, B, C and D. The data need to
+            match or be broadcastable into one ``shape``.
+        frequencies : array, double, None
+            Frequencies of the data in Hz. This is optional if using the
+            FrequencyData type for A, B, C, D.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import pyfar as pf
+        >>> frequencies = (100,200,300)
+
+        From np.array
+        >>> A = np.ones(len(frequencies))
+        >>> (B,C,D) = (A+1, A+2, A+3)
+        >>> tmat = pf.TransmissionMatrix.from_abcd(A,B,C,D, frequencies)
+
```

Comment:
```suggestion
    >>> tmat
    FrequencyData:
    (2, 2) channels with 3 frequencies
```
FrequencyData should be TransmissionMatrix, so we need to overwrite the ``FrequencyData.__repr__``